modules/ram-assumable-role/locals.tf

```diff
@@ -0,0 +1,47 @@
+locals {
+  role_name = var.role_name != "" ? var.role_name : substr("role-${random_integer.this.result}",0 ,32 )
```

Comment:
role_name = var.role_name != "" ? var.role_name : substr("ram-assumable-role-${random_integer.this.result}",0 ,32 )